aptos-move/block-executor/src/executor.rs

```diff
@@ -2074,6 +2075,11 @@ where
                 global_module_cache,
                 unsync_map.module_cache(),
             )?;
+            modules_published = true;
+        }
+        // For simplicity, flush layout cache on module publish.
+        if modules_published {
```

Comment:
This seems wasteful and I'm surprised we still can't do the selective flushing. (Layout wouldn't be the only info which can be outdated, anything we derive from code, like JIT, as well.) It's not relevant for this PR, but the objective to cache derived data is there since +2 years.

Reply:
It is about the representation: current layout representation does not work so we need to make sure it is such that 1) we can do selective flushing, 2) we do not make happy no-publish path suffer. Otherwise caching derived data should work, even now instead of flush we can iterate over layout cache, and for all layouts we check published ids and defining ids for layout intersection: can actually be done as part of this PR tbh